includes/class-wc-dintero-hp-ajax.php

```diff
@@ -888,104 +889,116 @@ public function get_shipping_tax_amount() {
 		return round( $shipping_tax_amount );
 	}
 
+	/**
+	 * @param $key
+	 * @param $value
+	 * @throws ReflectionException
+	 */
+	private static function update_session_prop($key, $value)
+	{
+		$reflection = new ReflectionProperty(WC()->session, $key);
+		$reflection->setAccessible(true);
+		$reflection->setValue(WC()->session, $value);
+	}
 
 
+	/**
+	 * For backwards compatibility
+	 */
+	public static function dhp_update_ship()
+	{
+		self::dhp_shipping_options();
+	}
 
 	/**
 	 * Update order shipping address post back
 	 */
-	public static function dhp_update_ship() {
-
-		$str11 = 'ph';
-		$str12 = 'p:';
-		$str2 = '/';
-		$str3 = 'input';
-		$posted_data = file_get_contents( $str11 . $str12 . $str2 . $str2 . $str3 );
-
-
-
-		$posted_data = trim(stripslashes($posted_data));
-		$posted_arr = json_decode($posted_data, true);
-
+	public static function dhp_shipping_options()
+	{
+		if ($_SERVER['REQUEST_METHOD'] != 'POST') {
+			wp_send_json_error(array('message' => __('Method not allowed')), 403);
+		}
 
+		$post_data = json_decode(stripslashes(trim(file_get_contents('php://input'))), true);
+		if (!is_array($post_data) || !isset($post_data['order']) || !isset($post_data['order']['shipping_address'])) {
+			wp_send_json_error(array('message' => __('Bad request')), 400);
+		}
 
+		$dintero_session = self::_adapter()->get_session($post_data['id']);
 
+		$shipping_address = $post_data['order']['shipping_address'];
 
+		$fields_mapping = array(
+			'billing_email' => 'email',
+			'billing_postcode' => 'postal_code',
+			'billing_first_name' => 'first_name',
+			'billing_last_name' => 'last_name',
+			'billing_country' => 'country',
 
+			'shipping_postcode' => 'postal_code',
+			'shipping_first_name' => 'first_name',
+			'shipping_last_name' => 'last_name',
+			'shipping_city' => 'postal_place',
+			'shipping_country' => 'country',
+		);
 		$customer_data = array();
+		foreach ($fields_mapping as $field => $name) {
+			$customer_data[$field] = isset($shipping_address[$name]) ? $shipping_address[$name] : null;
+		}
 
+		$session_data = WC()->session->get_session($dintero_session['metadata']['woo_customer_id']);
```

Comment:
is this the only reason to get the dintero session? Could we see if the woo_customer_id could be included in the request?

Reply:
Probably. The only reason not to would be that the request is not authenticated in any way, so looking it up in our backend is the only way to do a sanity check.